src/App.tsx

```diff
@@ -9,37 +9,55 @@ import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
 import { getUserPosts } from './api/posts';
-import { User } from './types/User';
-import { Post } from './types/Post';
+// import { Post } from './types/Post';
 import { Counter } from './features/counter/Counter';
+import { setAuthor } from './features/authorSlice';
+import { useAppDispatch, useAppSelector } from './app/hooks';
+import { RootState } from './app/store';
+import { User } from './types/User';
+import { setPosts, setLoaded, setHasError } from './features/postsSlice';
+import { setSelectedPost } from './features/selectedPostSlice';
 
 export const App: React.FC = () => {
-  const [posts, setPosts] = useState<Post[]>([]);
-  const [loaded, setLoaded] = useState(false);
-  const [hasError, setError] = useState(false);
+  const dispatch = useAppDispatch();
+
+  const author = useAppSelector(
+    (state: RootState) => state.author,
+  ) as User | null;
 
-  const [author, setAuthor] = useState<User | null>(null);
-  const [selectedPost, setSelectedPost] = useState<Post | null>(null);
+  // const [selectedPost, setSelectedPost] = useState<Post | null>(null);
```

Comment:
Remove comment. Try it write obvious code that explain all we need without comments)